src/coreclr/gc/gc.cpp

```diff
@@ -34216,6 +34217,9 @@ BOOL gc_heap::commit_mark_array_new_seg (gc_heap* hp,
             dprintf (GC_TABLE_LOG, ("partially in bgc range: seg %Ix-%Ix, bgc: %Ix-%Ix",
                                     start, end, lowest, highest));
             commit_flag = heap_segment_flags_ma_pcommitted;
+#ifdef USE_REGIONS
+            FATAL_GC_ERROR();
```

Comment:
should this be fatal, or just assert here?

Reply:
This was for testing in release build only, reverted to debug assert now.